src/ui/AccessibleLinksFormatter.cpp

```diff
@@ -0,0 +1,170 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file AccessibleLinksFormatter.h
+ @brief Define a helper class to format text with link in a way, accessible to VI users.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "AccessibleLinksFormatter.h"
+
+#include "ShuttleGui.h"
+
+#include <unordered_map>
+#include <algorithm>
+
+#include <wx/hyperlink.h>
+
+
+namespace
+{    
+size_t OffsetPosition(size_t position, size_t length)
+{
+   if (position == wxString::npos)
+      return wxString::npos;
+
+   return position + length;
+}
+}
+
+AccessibleLinksFormatter::AccessibleLinksFormatter(TranslatableString message)
+    : mMessage(std::move(message))
+{
+}
+
+AccessibleLinksFormatter& AccessibleLinksFormatter::FormatLink(
+   wxString placeholder, TranslatableString value, std::string targetURL)
+{
+   mFormatArguments.push_back({ 
+       std::move(placeholder), 
+       std::move(value), 
+       {}, 
+       std::move(targetURL)
+   });
+
+   return *this;
+}
+
+AccessibleLinksFormatter& AccessibleLinksFormatter::FormatLink(
+   wxString placeholder, TranslatableString value,
+   LinkClickedHandler handler)
+{
+   mFormatArguments.push_back({ 
+       std::move(placeholder), 
+       std::move(value), 
+       std::move(handler), 
+       {} 
+   });
+
+   return *this;
+}
+
+void AccessibleLinksFormatter::Populate(ShuttleGui& S)
+{
+   // Just add the text, if there are no links to format
+   if (mFormatArguments.empty())
+   {
+      S.AddFixedText(mMessage);
+      return;
+   }
+
+   wxString translated = mMessage.Translation();
+
+   ProcessArguments(translated);
+
+   const int borderSize = S.GetBorder();
+
+   S.Prop(0).StartPanel();
+   S.StartWrapLay();
+   {
+      size_t currentPosition = 0;
+
+      S.SetBorder(0);
+
+      if (borderSize > 0)
+         S.AddSpace(borderSize);
```

Comment:
I see that this explains why wxWrapSizer was not constructed with `wxREMOVE_LEADING_SPACES`